analytics/location_heatmaps/geo_utils.py

```diff
@@ -278,141 +296,124 @@ def split_regions(tree_prefix_list,
-            #       count)
-          else:
-            conf_int = 1.96 * np.sqrt((p * (1 - p) / last_count)) * last_count
-            cond = conf_int < threshold
-            intervals.append(conf_int)
-            # print(last_prefix, prefix, last_prefix_pos, last_count, count, conf_int, cond)
+  for i in range(len(tree_prefix_list)):
+    if count_min:
+      count = count_min.query(tree_prefix_list[i])
+    else:
+      count = vector_counts[i] if not expand_all else np.inf
+    prefix = tree_prefix_list[i]
+
+    # check whether the tree has reached the bottom
+    if len(prefix.split('/')) >= image_bit_level:
+      continue
+    if last_result:
+      cond = create_confidence_interval_condition(last_result, prefix, count,
+                                                  split_threshold)
+    else:
+      cond = count > split_threshold
+    if expand_all or cond:
+      for child in DEFAULT_CHILDREN:
+        new_prefix = f'{prefix}/{child}'
+        fresh_expand += update_tree(new_prefix, new_tree, new_tree_prefix_list)
+    else:
+      if collapse_threshold is not None and \
+        count <= collapse_threshold and len(prefix) > 2:
+        old_prefix = prefix[:-3]
+        collapsed += 1
+        created += update_tree(old_prefix, new_tree, new_tree_prefix_list)
       else:
-        cond = count > threshold
-      # print(cond, threshold, count)
-      if cond:
-        for child in DEFAULT_CHILDREN:
-          new_prefix = f'{prefix}/{child}'
-          if not new_tree.has_key(new_prefix):
-            fresh_expand += 1
-            new_tree[new_prefix] = len(new_tree_prefix_list)
-            new_tree_prefix_list.append(new_prefix)
+        unchanged += update_tree(prefix, new_tree, new_tree_prefix_list)
+
+  return new_tree, new_tree_prefix_list, fresh_expand
+
+
+def split_regions_aux(tree_prefix_list,
+                  vector_counts,
+                  split_threshold,
+                  image_bit_level,
+                  collapse_threshold=None,
+                  expand_all=False,
+                  last_result: AlgResult = None,
+                  count_min=None):
+  """Use expansion with aux data.
+
+  We check both counts for positive and negative attributes for each location.
+
+  Args:
+      tree_prefix_list: matches vector id to the tree prefix.
+      vector_counts: vector values aggregated from the users.
+      split_threshold: threshold value used to split the nodes.
+      image_bit_level: stopping criteria once the final resolution is reached.
+      collapse_threshold: threshold value used to collapse the nodes.
+      expand_all: expand all regions,
+      last_result: use previous level results to compute conf intervals,
+      count_min: use count-min sketch
+  Returns:
+      new_tree, new_tree_prefix_list, fresh_expand
+  """
+  new_tree_prefix_list = list()
+  new_tree = pygtrie.StringTrie()
+  collapsed = 0
+  created = 0
+  fresh_expand = 0
+  unchanged = 0
```

Comment:
likewise re. collapsed, created, and unchanged being unused

Reply:
done